ui/js/models/resultsets_store.js

```diff
@@ -202,16 +202,13 @@ treeherder.factory('ThResultSetStore', [
                     sortGroupedJobs,
                     groupJobByPlatform
                 );
-                return sortAndGroupJobs(resultSet.jobList).
-                    then(function(groupedJobs){
-                        _.extend(resultSet, groupedJobs);
-                        mapPlatforms(
-                            repoName,
-                            resultSet
-                        );
-                        updateUnclassifiedFailureCountForTiers(repoName);
-                        $rootScope.$emit(thEvents.applyNewJobs, resultSetId);
-                    });
+                _.extend(resultSet, sortAndGroupJobs(resultSet.jobList));
+                mapPlatforms(
+                    repoName,
+                    resultSet
+                );
+                updateUnclassifiedFailureCountForTiers(repoName);
+                $rootScope.$emit(thEvents.applyNewJobs, resultSetId);
             }
             return $q.defer().resolve();
```

Comment:
I don't think this ``$q.defer().resolve()`` is needed any longer.  It was only there in case the ``jobList.length`` was 0.  The last place that returns this expecting it to be a promise is line 1010.  But I think you could just call it, since it's synchronous now.